### ISSUE TITLE
MySQL errors

### ISSUE THREAD

**Issue by netniV** [Cacti/cacti#1187]
As part of another issue I was looking into, I had a reason to look at the MySQL logs where I saw a large number of `timeout reading communication packets`.  If I was operating over the network I would expect this, but I'm operating locally.

This was mentioned during issue #902 but I am not sure was addressed.  Looking only at https://www.percona.com/blog/2016/05/16/mysql-got-an-error-reading-communication-packet-errors/ would suggest that the connection isn't being closed properly. 

```
2018-01-09T06:03:09.123220Z 1915977 [Note] Aborted connection 1915977 to db: 'cacti_test' user: 'cacti_test' host: 'localhost' (Got timeout reading communication packets)
2018-01-09T06:06:23.621540Z 1899847 [Note] Aborted connection 1899847 to db: 'cacti_test' user: 'cacti_test' host: 'localhost' (Got timeout reading communication packets)
2018-01-09T06:06:33.445624Z 1912280 [Note] Aborted connection 1912280 to db: 'cacti_test' user: 'cacti_test' host: 'localhost' (Got timeout reading communication packets)
2018-01-09T06:06:47.365485Z 2023916 [Note] Aborted connection 2023916 to db: 'cacti_test' user: 'cacti_test' host: 'localhost' (Got timeout reading communication packets)
2018-01-09T06:07:27.094078Z 1916371 [Note] Aborted connection 1916371 to db: 'cacti_test' user: 'cacti_test' host: 'localhost' (Got timeout reading communication packets)
2018-01-09T06:07:33.362718Z 1900539 [Note] Aborted connection 1900539 to db: 'cacti_test' user: 'cacti_test' host: 'localhost' (Got timeout reading communication packets)
```

I am assuming that this will be the poller or spine as, whilst I know I am on a lot, I was actually sleeping at this point! :)

**Comment by cigamit** [comment]
Which poller?

**Comment by netniV** [comment]
I am guessing CMD.PHP because I haven't changed the test system at all and that's the default.

**Comment by cigamit** [comment]
We are using persistent connections, so a close does nothing.  Eventually they are closed out on their own if not used over a period of time.  Could be a plugin though that does not include bottom_footer.php, which, by the way, includes a db_close() statement, or a die(), or an exit() somewhere in the CLI's.  We might want to consider a cacti_exit() wrapper in place of using exit() so we can close the connections.

There are a lot of exits.  Don't kill yourself.  Roney should comment on this one too.

**Comment by YATV** [comment]
Hello!
I have the same error in log mysqld:

```
2018-02-12T05:10:00.347857Z 9506 [Note] Aborted connection 9506 to db: 'base-monitoring' user: 'baseuser' host: '10.10.10.8' (Got an error reading communication packets)
2018-02-12T05:10:00.445754Z 9517 [Note] Aborted connection 9517 to db: 'base-monitoring' user: 'baseuser' host: '10.10.10.8' (Got an error reading communication packets)
2018-02-12T05:15:00.218505Z 9653 [Note] Aborted connection 9653 to db: 'base-monitoring' user: 'baseuser' host: '10.10.10.8' (Got an error reading communication packets)
2018-02-12T05:15:00.319289Z 9664 [Note] Aborted connection 9664 to db: 'base-monitoring' user: 'baseuser' host: '10.10.10.8' (Got an error reading communication packets)
```

I spent a lot of time to solve it.
And I corrected this mistake !!!
The problem with the name of the database.
If the name of the database has the characters "-" or "_" (other possible). Then we get this in logs:

```
2018-02-11 16:40:00 - SPINE: Poller[1] ERROR: SS[0] Script Server did not start properly return message was:
 'The Cacti Database has not been initialized.  Please initilize it before continuing.
To initilize the Cacti database, issue the following commands either as root or using a valid account.
  mysqladmin -uroot -p create cacti
  mysql -uroot -p -e "grant all on cacti.* to 'someuser'@'localhost' identified by 'somepassword'"
  mysql -uroot -p -e "grant select on mysql.time_zone_name to 'someuser'@'localhost' identified by 'somepassword'"
  mysql -uroot -p cacti < /pathcacti/cacti.sql
Where <b>/pathcacti/</b> is the path to your Cacti install location.
Change <b>someuser</b> and <b>somepassword</b> to match your site preferences.  The defaults are <b>cactiuser</b> for both user and password.
<b>NOTE:</b> When installing a remote poller, the <b>config.php</b> file must be writable by the Web Server account, and must include valid connection information to the main Cacti server.  The file should be changed to read only after the install is completed.
'2018-02-11 16:40:00 - SPINE: Poller[1] ERROR: SS[0] PHP Script Server communications lost.  Restarting PHP Script Server
2018-02-11 16:40:00 - SPINE: Poller[1] ERROR: SS[0] Script Server did not start properly return message was: 'The Cacti Database has not been initialized.  Please initilize it before continuing.
To initilize the Cacti database, issue the following commands either as root or using a valid account.
  mysqladmin -uroot -p create cacti
  mysql -uroot -p -e "grant all on cacti.* to 'someuser'@'localhost' identified by 'somepassword'"
  mysql -uroot -p -e "grant select on mysql.time_zone_name to 'someuser'@'localhost' identified by 'somepassword'"
  mysql -uroot -p cacti < /pathcacti/cacti.sql
Where <b>/pathcacti/</b> is the path to your Cacti install location.
Change <b>someuser</b> and <b>somepassword</b> to match your site preferences.  The defaults are <b>cactiuser</b> for both user and password.
<b>NOTE:</b> When installing a remote poller, the <b>config.php</b> file must be writable by the Web Server account, and must include valid connection information to the main Cacti server.  The file should be changed to read only after the install is completed.

```

It is treated by renaming the database.
But it's right - it's to quote the  \`base-name\` 
Programmers correct this stupid mistake, please.
Thank you

**Comment by cigamit** [comment]
This note should be on a separate thread/bug.  Can you open a new bug since its a different issue?  Make the bug title "Installer should block installation if database name includes special characters".

**Comment by netniV** [comment]
Sorry, I'm confused.  What's a different issue?

**Comment by cigamit** [comment]
This post was caused by insomnia.  Going to try to get back to sleep now ;)  I posted to the wrong ticket :)

**Comment by cigamit** [comment]
The user deleted his/her comment.  I deleted a few more ;) 1.1.35 is out.  Hoping to focus back again on 1.2 now.

**Comment by cigamit** [comment]
After the bottom_footer.php changes in 1.1.36, I think we should close this and see what happens in a production environment.  We could get something like this in a dev environment if your page has some syntax error for example.
